libs/ledger-live-common/src/families/evm/prepareTransaction.ts

```diff
@@ -57,33 +81,39 @@ export const prepareTokenTransaction = async (
   tokenAccount: TokenAccount,
   typedTransaction: EvmTransaction
 ): Promise<EvmTransaction> => {
+  const [recipientErrors] = validateRecipient(account, typedTransaction);
   const amount = typedTransaction.useAllAmount
     ? tokenAccount.balance
     : typedTransaction.amount;
-  const protoTransaction = {
-    ...typedTransaction,
-    amount: new BigNumber(0),
-  };
-  const [recipientErrors] = validateRecipient(account, protoTransaction);
-  const [amountErrors] = validateAmount(tokenAccount, protoTransaction, amount);
-  if (Object.keys(amountErrors).length || Object.keys(recipientErrors).length) {
-    return typedTransaction;
-  }
-
-  const data = getTransactionData({ ...typedTransaction, amount });
+  const data = !Object.keys(recipientErrors).length
+    ? getTransactionData({ ...typedTransaction, amount })
```

Comment:
It seems `getTransactionData` can throw, but are we expecting that it should never fail here?

Reply:
If `getTransactionData` fails, it means we gave wrong parameters (types) to the smart contract ABI, so it should definitely throw and that error should be visible to the user. I don't think that should ever happened though 🤔